modules/messages/server/services/message-to-influx.server.service.js

```diff
@@ -0,0 +1,192 @@
+      // already (at least the one just saved))
+      if (!firstMessage) {
+        var err = new Error('first message not found, but should have been already saved');
+        done(err);
+      }
+
+
+      // is the new message the first message of the thread?
+      isFirstMessage = String(firstMessage._id) === String(message._id);
+
+      // can the message be the actual first reply?
+      // - is it not the firstMessage?
+      // is the sender and receiver in different order than in the firstMessage?
+      var canBeTheFirstReply = !isFirstMessage && String(firstMessage.userTo) === String(userFrom);
+      // if this can be the oldest reply, find the oldest reply of the thread
+      if (canBeTheFirstReply) {
+        return Message.findOne({
+          userTo: firstMessage.userFrom,
+          userFrom: firstMessage.userTo
+        })
+        .sort({ created: 1 })
+        .exec(function (err, firstReply) {
+          return done(err, firstMessage, firstReply);
+        });
+      } else {
+        return done(null, firstMessage, null);
+      }
+    },
+
+    function prepareData(firstMessage, firstReply, done) {
+      // is the new message the first reply of the thread?
+      isFirstReply = !!(firstReply && String(firstReply._id) === String(message._id));
+
+      // count the reply time for statistics (miliseconds)
+      var replyTime;
+      if (isFirstReply) {
+        replyTime = firstReply.created.getTime() - firstMessage.created.getTime();
+      }
+
+      // count length of the message
+      // excluding html tags and multiple whitespace characters
+      var msgLen = textProcessor.plainText(message.content, true).length;
+
+
+      // message position in the thread
+      var position;
+      if (isFirstMessage) {
+        position = 'first';
+      } else if (isFirstReply) {
+        position = 'first_reply';
+      } else {
+        position = 'other';
+      }
+
+      var msgLenType = msgLen < config.longMessageMinimumLength ? 'short' : 'long';
+
+      // values for influxdb, using camelCase for tag and field keys
+      var fields = {
+        messageId: String(message._id), // id of message, (discussed keeping for now)
+        idFrom: String(userFrom), // id of sender
+        idTo: String(userTo), // id of receiver
+        messageLength: msgLen, // length of the content
+        time: message.created.getTime() // creation timestamp (milliseconds)
+      };
+
+      // we measure the reply time only for the first replies (time since the
+      // first message sent by the other user)
+      if (isFirstReply) {
+        fields.replyTime = replyTime;
+      }
+
+      // tags for influxdb
+      var tags = {
+        position: position, // position (first, first_reply, normal)
+        messageLengthType: msgLenType // (short, long) content (shortness defined in a config)
+      };
+
+      return done(null, fields, tags);
+    }
+  ], callback);
```

Comment:
This gets called with error from line [103](https://github.com/mrkvon/trustroots/blob/f051c9649412ee69ac72ecd1274e2d6ff21510fd/modules/messages/server/services/message-to-influx.server.service.js#L103) `done(err)` so I suggest wrapping it with this:
```javascript
function(err) { 
  if (err) {
    console.error(err);
    return;
  }
  callback();
}
```